custom_metrics/markup.js

```diff
@@ -0,0 +1,647 @@
+      }
+    })(),
+
+    // dir attributes
+    // Used by Markup, 2019/09_28
+    'dirs': (() => {   
+      try {   
+        let result = {html_dir: null};
+
+        function findDirs(selector) {
+          let target = {values: {}};
+          var nodes = [...document.querySelectorAll(selector)];
+
+          target.total = nodes.length;
+    
+          nodes.forEach((n) => {
+            let dir = n.getAttribute("dir");
+    
+            if (dir) {
+              if (target.values[dir])
+                target.values[dir]++;
+              else 
+                target.values[dir] = 1;
+            }
+    
+          });
+          return target;
+        };
+
+        let dir = document.querySelector('html[dir]')?.getAttribute("dir");
+
+        if (dir === "" || dir)
+          result.html_dir = dir;
+        
+        result.body_nodes_dir = findDirs('body *[dir]');
+
+        return result;
+      }
+      catch(e) {
+        return logError("dirs", e);
+      }
+    })(),
+
+    // input tags
+    // Used by Markup, 2019/09_28
+    'inputs': (() => {   
+      try {   
+        let result = {types: {}};
+
+        var nodes = [...document.querySelectorAll('input')];
+
+        result.total = nodes.length;
+
+        nodes.forEach((n) => {
+          let type = n.getAttribute("type");
+
+          if (type) {
+              if (result.types[type])
+                result.types[type]++;
+              else 
+                result.types[type] = 1;
+          }
+        });
+
+        return result;
+      }
+      catch(e) {
+        return logError("inputs", e);
+      }
+    })(),
+
+    // script tags
+    // Used by SEO, 2019/09_28
+    'scripts': (() => {   
+      try {   
+        let result = {types: {}, inline: 0, src: 0};
+
+        var nodes = [...document.querySelectorAll('script')];
+
+        result.total = nodes.length;
+
+        nodes.forEach((n) => {
+          let type = n.getAttribute("type");
+
+          if (type) {
+              if (result.types[type])
+                result.types[type]++;
+              else 
+                result.types[type] = 1;
+          }
+
+          let src = n.getAttribute("src");
+
+          if (src)
+            result.src++;
+          else 
+            result.inline++;
+        });
+
+        return result;
+      }
+      catch(e) {
+        return logError("scripts", e);
+      }
+    })(),
+
+    // video tags
+    // Used by Markup
+    'videos': (() => {
+      try {
+        let result = {autoplay: {}};
+
+        const nodes = document.querySelectorAll('video');
+
+        nodes.forEach((n) => {
+          let autoplay = n.getAttribute("autoplay");
+
+          if (result.autoplay[autoplay])
+            result.autoplay[autoplay]++;
+          else 
+            result.autoplay[autoplay] = 1;
+          });
+
+        result.total = nodes.length;
+
+        return result;
+      }
+      catch(e) {
+        return logError("videos", e);
+      }
+    })(),
```

Comment:
The `videos` metric in #172 provides the same info on video elements and more, so we can remove this one.
```suggestion
```

Reply:
ok